pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -1346,35 +1346,29 @@ func OwnerRef(ro runtime.Object, scheme *runtime.Scheme) (metav1.OwnerReference,
 	}, nil
 }
 
-func (r *ReconcilePerconaServerMongoDB) hasMongodCustomConfiguration(cr *api.PerconaServerMongoDB, rs *api.ReplsetSpec) (bool, error) {
-	if rs.Configuration != "" {
-		return true, nil
+func (r *ReconcilePerconaServerMongoDB) getCustomConfigurationSource(namespace, name string) (psmdb.VolumeSourceType, error) {
+	var n = types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
 	}
 
-	return r.existsConfigMap(cr.Namespace, psmdb.MongodConfigCMName(cr.Name, rs.Name))
-}
-
-func (r *ReconcilePerconaServerMongoDB) hasMongosCustomConfiguration(cr *api.PerconaServerMongoDB) (bool, error) {
-	if cr.Spec.Sharding.Mongos.Configuration != "" {
-		return true, nil
+	err := r.client.Get(context.Background(), n, &corev1.Secret{})
```

Comment:
can we move this part to function?
Because in future we could add more sources and have to duplicate this code again

Reply:
did I get you right? 27ce082